substrate/frame/asset-conversion/src/lib.rs

```diff
@@ -789,14 +899,14 @@ pub mod pallet {
 			amount: T::AssetBalance,
 			keep_alive: bool,
 		) -> Result<T::AssetBalance, DispatchError> {
-			let result = match T::MultiAssetIdConverter::try_convert(asset_id) {
+			let preservation = match keep_alive {
+				true => Preserve,
+				false => Expendable,
+			};
+			match T::MultiAssetIdConverter::try_convert(asset_id) {
 				MultiAssetIdConversionResult::Converted(asset_id) =>
-					T::Assets::transfer(asset_id, from, to, amount, Expendable),
+					T::Assets::transfer(asset_id, from, to, amount, preservation),
```

Comment:
So `keep_alive` didn't work previously?

Reply:
@liamaharon yes, it did not.
@gilescope the transfer happens only for assets moving in the middle of the swap (does not work on the edges), and for this transfer we have a test, last case of `swap_when_existential_deposit_would_cause_reaping_pool_account`